bfvmm/include/intrinsics/x86/intel/apic/x2apic.h

```diff
@@ -2112,7 +2334,30 @@ struct EXPORT_X2APIC x2apic_control final : public lapic_control
     // @note to ensure an accurate result, the caller should mask
     // the vector prior to the call
     //
-    bool level_triggered(vector_type vec) noexcept override;
+    bool level_triggered(vector_type vec) noexcept override
+    {
```

Comment:
rather than dividing and % use
auto reg = vec & 0xE0
auto bit = 1ULL << (vec & 0x1F)

switch (reg) {
    case 0: return <msr>::get() & bit;
   ...